### PR TITLE
Global: Replace with GCS_SEND_TEXT

### PR DESCRIPTION
**Pull request by muramura**
If a peripheral is selected, change the definition to one that does not notify the GCS of the message.

### CODE REVIEW

File: Tools/AP_Periph/AP_Periph.cpp [hunk 1]
```diff
@@ -113,7 +113,7 @@ void AP_Periph_FW::init()
 #if HAL_GCS_ENABLED
     gcs().setup_console();
     gcs().setup_uarts();
-    gcs().send_text(MAV_SEVERITY_INFO, "AP_Periph GCS Initialised!");
+    GCS_SEND_TEXT(MAV_SEVERITY_INFO, "AP_Periph GCS Initialised!");
```
Comment (IamPete1) [inline Tools/AP_Periph/AP_Periph.cpp:116]: This is harmless, but also incorrect.

File: libraries/AC_AutoTune/AC_AutoTune.cpp [hunk 2]
```diff
@@ -601,22 +601,22 @@ void AC_AutoTune::update_gcs(uint8_t message_id) const
 {
     switch (message_id) {
     case AUTOTUNE_MESSAGE_STARTED:
-        gcs().send_text(MAV_SEVERITY_INFO,"AutoTune: Started");
+        GCS_SEND_TEXT(MAV_SEVERITY_INFO,"AutoTune: Started");
```
Comment (IamPete1) [inline libraries/AC_AutoTune/AC_AutoTune.cpp:604]: This is a example where the whole function should be defined out.

**Comment by IamPete1** [comment]
I think we need a little more thought than a global replace. In some cases this is correct, in other cases this will build but it won't necessarily do the correct thing.

File: Tools/AP_Periph/AP_Periph.cpp (hunk 1 above)
Comment (peterbarker) [reply Tools/AP_Periph/AP_Periph.cpp:116]: Not sure about "incorrect" here.  "strange", yes :-)

File: libraries/AC_AutoTune/AC_AutoTune.cpp (hunk 2 above)
Comment (peterbarker) [reply libraries/AC_AutoTune/AC_AutoTune.cpp:604]: Scope creep :-)

**Comment by peterbarker** [comment]
> I think we need a little more thought than a global replace. In some cases this is correct, in other cases this will build but it won't necessarily do the correct thing.

I'm not sure how it can not do the correct thing.

I actually like this PR - it sets the pattern everywhere so we don't get more instances of it happening.

I've also increasingly been musing on how we go via the GCS library for the "send text" functionality, except on Periph where it gets diverted. 

It wends itself out of the system via various RC telemetry libraries, so when we want to do RC telemetry on Periph we might find we have to include the GCS library, which would be bad.  If everything goes via the GCS_SEND_TEXT macro then we might be able to easily split out a separate library which handles the text-to-operator queue.

**Comment by IamPete1** [comment]
> I'm not sure how it can not do the correct thing.

What I mean is that even it this builds it does not mean that we would want to be spamming the message out over CAN. In some places there needs to be a little more thought. This change allows us to skip putting in that extra thought. Maybe that OK, but it makes it harder to come back and put in that effort later. The AutoTune lib is the case in point. Do we ever want to run that without a GCS? Maybe with a different telem system in the future, but then do we want to send the same number of messages?

My preference would be to fix these as they actually cause build issues so we can put in that extra effort. 

Some of these I have no issue with, compass, rangefinder ect.

File: ArduCopter/surface_tracking.cpp [hunk 3]
```diff
@@ -100,12 +100,16 @@ void Copter::SurfaceTracking::set_surface(Surface new_surface)
     }
     // check we have a range finder in the correct direction
     if ((new_surface == Surface::GROUND) && !copter.rangefinder.has_orientation(ROTATION_PITCH_270)) {
+#if !defined(HAL_BUILD_AP_PERIPH)
```
Comment (tridge) [inline ArduCopter/surface_tracking.cpp:103]: this doesn't make sense

File: ArduPlane/mode_fbwa.cpp [hunk 4]
```diff
@@ -30,7 +30,9 @@ void ModeFBWA::update()
         if (tdrag_mode && !plane.auto_state.fbwa_tdrag_takeoff_mode) {
             if (plane.auto_state.highest_airspeed < plane.g.takeoff_tdrag_speed1) {
                 plane.auto_state.fbwa_tdrag_takeoff_mode = true;
+#if !defined(HAL_BUILD_AP_PERIPH)
```
Comment (tridge) [inline ArduPlane/mode_fbwa.cpp:33]: this doesn't make sense in plane

**Review by tridge — CHANGES_REQUESTED** [review]
i'd like to drop the vehicle specific changes just to minimise conflicts with other PRs

File: ArduCopter/surface_tracking.cpp (hunk 3 above)
Comment (muramura) [reply ArduCopter/surface_tracking.cpp:103]: I would go without this change.

File: ArduPlane/mode_fbwa.cpp (hunk 4 above)
Comment (muramura) [reply ArduPlane/mode_fbwa.cpp:33]: I would go without this change.

**Comment by rmackay9** [comment]
@muramura,

Can  you rebase this on master?

**Comment by rmackay9** [comment]
Also remember to remove the vehicle specific changes.

**Comment by peterbarker** [comment]
Replaced by #27776

**Pull request closed without merging**